stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -549,6 +561,19 @@ internal func dumpCTensorHandleContent(
   }
 }
 
+/// Given an input like "foo.tf_13" (prefix if a SIL graph function like
+/// "foo.tf_13_CPU.device_partition"), extract and return `13` as the device
+/// index.
+internal func decodeDeviceIndexStr(_ opTypePrefix: Substring) -> Substring {
```

Comment:
1. Use `private` when it's not used in other files.
2. It's only used within `TFEState.init`, so why not define this function in there? You can define a function using the `func` syntax anywhere in Swift. That way, you will also be able to drop the access modifier.
3. The function is not side-effecting and thus should not use a bare-form verb.
4. "Index" is a bit unclear if we named the function `deviceIndex(from:)` because it is not returning an integer. So we need to clarify that in the function name.

As such, I would recommend the following name:
```suggestion
private func deviceIndexSubstring(from opTypePrefix: Substring) -> Substring {
```

Reply:
Ok! When the function is defined nested in `init()` so naming doesn't matter that much anymore because it's local enough!

Reply:
Ah I overlooked the suggested func name. It **is** a better name, and adopted it now. Thanks!